sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -1081,7 +1084,20 @@ public BigtableWriteException(KV<ByteString, Iterable<Mutation>> record, Throwab
    * @see com.google.cloud.bigtable.config.BigtableVersionInfo
    */
   private static String getBeamSdkPartOfUserAgent() {
-    ReleaseInfo info = ReleaseInfo.getReleaseInfo();
+    // Attempt to get the Dataflow distribution release information, if we fail, fallback to using
+    // the Apache Beam release information.
+    ReleaseInfo info;
+    try {
+      Class<?> kls = ReflectHelpers.findClassLoader().loadClass(
+          "org.apache.beam.runners.dataflow.DataflowRunnerInfo");
```

Comment:
<!--new_thread; commit:e9176e9939f0dd61789718203cca95ed760e0954; resolved:1-->
This is gross, but I see why it has to happen, short term. Perhaps this is best delivered via PipelineOptions or some such, which can now be accessed via new DoFn.

Reply:
I'll revert this one as now I don't believe BigtableIO should care whether its Dataflow/Apache Beam distribution and if we need to patch BigtableIO, we can patch the user agent.